hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java

```diff
@@ -135,17 +150,47 @@ private TransactionMetadata preHandle(
             // TODO: Verify signature via the platform once this functionality was implemented
 
             // 5. Return TransactionMetadata
-            return metadata;
+            return createTransactionMetadata(statesTracker.getUsedStates(), handlerContext);
 
         } catch (PreCheckException preCheckException) {
-            return new ErrorTransactionMetadata(
-                    txBody, preCheckException.responseCode(), preCheckException);
+            return new TransactionMetadata(txBody, payerID, preCheckException.responseCode());
```

Comment:
We should probably use `createInvalidTransactionMetadata` or so, to be consistent with `createTransactionMetadata`

Reply:
Ok, will change it.